tests/client/handlers/nodes.test.ts

```diff
@@ -680,3 +694,266 @@ describe('nodesPing', () => {
+    );
+    const keysPath = path.join(dataDir, 'keys');
+    keyRing = await KeyRing.createKeyRing({
+      password,
+      keysPath,
+      passwordOpsLimit: keysUtils.passwordOpsLimits.min,
+      passwordMemLimit: keysUtils.passwordMemLimits.min,
+      strictMemoryLock: false,
+      logger,
+    });
+    tlsConfig = await testsUtils.createTLSConfig(keyRing.keyPair);
+    const dbPath = path.join(dataDir, 'db');
+    db = await DB.createDB({
+      dbPath,
+      logger,
+    });
+    sigchain = await Sigchain.createSigchain({
+      db,
+      keyRing,
+      logger,
+    });
+    nodeGraph = await NodeGraph.createNodeGraph({
+      db,
+      keyRing,
+      logger: logger.getChild('NodeGraph'),
+    });
+    taskManager = await TaskManager.createTaskManager({
+      db,
+      logger,
+      lazy: true,
+    });
+    nodeConnectionManager = new NodeConnectionManager({
+      keyRing,
+      nodeGraph,
+      // TLS not needed for this test
+      tlsConfig: {} as TLSConfig,
+      connectionConnectTimeoutTime: 2000,
+      connectionIdleTimeoutTime: 2000,
+      logger: logger.getChild('NodeConnectionManager'),
+    });
+    nodeManager = new NodeManager({
+      db,
+      keyRing,
+      nodeConnectionManager,
+      nodeGraph,
+      sigchain,
+      taskManager,
+      gestaltGraph: {} as GestaltGraph,
+      logger,
+    });
+    await nodeConnectionManager.start({ host: localhost as Host });
+    await taskManager.startProcessing();
+    clientService = new ClientService({
+      tlsConfig,
+      logger: logger.getChild(ClientService.name),
+    });
+    await clientService.start({
+      manifest: {
+        nodesGetAll: new NodesGetAll({
+          nodeGraph,
+          keyRing,
+        }),
+      },
+      host: localhost,
+    });
+    webSocketClient = await WebSocketClient.createWebSocketClient({
+      config: {
+        verifyPeer: false,
+      },
+      host: localhost,
+      logger: logger.getChild(WebSocketClient.name),
+      port: clientService.port,
+    });
+    rpcClient = new RPCClient({
+      manifest: {
+        nodesGetAll,
+      },
+      streamFactory: () => webSocketClient.connection.newStream(),
+      toError: networkUtils.toError,
+      logger: logger.getChild(RPCClient.name),
+    });
+  });
+  afterEach(async () => {
+    mockedPingNode.mockRestore();
+    await taskManager.stopProcessing();
+    await taskManager.stopTasks();
+    await clientService.stop({ force: true });
+    await webSocketClient.destroy({ force: true });
+    await sigchain.stop();
+    await nodeGraph.stop();
+    await nodeConnectionManager.stop();
+    await db.stop();
+    await keyRing.stop();
+    await taskManager.stop();
+    await fs.promises.rm(dataDir, {
+      force: true,
+      recursive: true,
+    });
+  });
+  test('gets all nodes', async () => {
+    mockedPingNode.mockResolvedValue(false);
+    const response = await rpcClient.methods.nodesGetAll({});
+    expect(response).toBeTruthy();
+  });
```

Comment:
You shouldn't really copy the nodes ping tests wholesale. Make sure you're only using the dependencies you need. We don't need mocking for the ping node here.

Also, test more than there is a response. Make sure the expected data is correct as well.

Same needs to be done for the nodes connections test below.